### PR TITLE
#187419169 user can login via email and password

### PR DESCRIPTION
**Pull request by niyobertin**
Have a user login functionality into E-commerce application via email and password, unit test for login endpoint and swagger docummentation for login endpoint

**Description of Task to be completed?**

Have the following endpoint:
POST  /api/v1/login

**How should this be manually tested?**

Clone[ this ](https://github.com/atlp-rwanda/eagles-ec-be)repository , cd into it and run:

- _npm run dev_  to start server . while server running,  use postman and send

1.POST request on  locahost:PORT/api/v1/users/login by providing email and password. 
 it will return jwt as result of successful login.
2.  in your browser send request on _localhost:PORT/docs_ --->swagger documentation.

- _npm run test_  for unit test

**What are the relevant pivotal tracker/Trello stories?**

#187419169 

**Screenshots**
the screenshot below show a successful login. on left side there is a user credetiols(email & password) and right side response with jwt.
success

### CODE REVIEW

**Review by Heisjabo — APPROVED** [review]

**Review by yvanddniyo — APPROVED** [review]
LGTM

**Review by soleil00 — APPROVED** [review]
nice work

**Review by MugemaneBertin2001 — APPROVED** [review]
LGTM

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,4 @@
 PORT = 3000
 DB_CONNECTION = "" --> TODO: put your own connection string here
+TEST_DB = postgres://postgres:test@123@localhost:3000/testing
+JWT_SECRET = 4jDzNz7RW1BtVWRCauURswYzRaMo4z1C
```
Comment (teerenzo) [inline .env.example]: remove values

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Review by teerenzo — APPROVED** [review]

**Pull request merged**